src/SFML/Audio/SoundStream.cpp

```diff
@@ -395,6 +507,14 @@ bool SoundStream::getLoop() const
 }
 
 
+////////////////////////////////////////////////////////////
+void SoundStream::setEffectProcessor(EffectProcessor effectProcessor)
+{
+    m_impl->effectProcessor = std::move(effectProcessor);
+    m_impl->connectEffect(!!m_impl->effectProcessor);
```

Comment:
```suggestion
    m_impl->connectEffect(static_cast<bool>(m_impl->effectProcessor));
```

Personal preference, a bit longer but easier to understand if not familiar with `!!` idiom